tests/phpunit/tests/interactivity-api/wpInteractivityAPI.php

```diff
@@ -508,50 +508,130 @@ public function test_process_directives_doesnt_change_html_if_contains_unbalance
 	}
 
 	/**
-	 * Tests that the `process_directives` returns the same HTML if it finds an
-	 * SVG tag.
+	 * Tests that the `process_directives` process the HTML outside a SVG tag.
 	 *
-	 * @ticket 60356
+	 * @ticket 60517
 	 *
 	 * @covers ::process_directives
 	 */
-	public function test_process_directives_doesnt_change_html_if_contains_svgs() {
-		$this->interactivity->state( 'myPlugin', array( 'id' => 'some-id' ) );
+	public function test_process_directives_changes_html_if_contains_svgs() {
+		$this->interactivity->state(
+			'myPlugin',
+			array(
+				'id'    => 'some-id',
+				'width' => '100',
+			)
+		);
 		$html           = '
-			<div data-wp-bind--id="myPlugin::state.id">
-				<svg height="100" width="100">
+			<header>
+				<svg height="100" data-wp-bind--width="myPlugin::state.width">
 					<circle cx="50" cy="50" r="40" stroke="black" stroke-width="3" fill="red" />
-				</svg> 
-			</div>
+				</svg>
+				<div data-wp-bind--id="myPlugin::state.id"></div>
+				<div data-wp-bind--id="myPlugin::state.width"></div>
+			</header>
 		';
 		$processed_html = $this->interactivity->process_directives( $html );
 		$p              = new WP_HTML_Tag_Processor( $processed_html );
-		$p->next_tag();
+		$p->next_tag( 'svg' );
+		$this->assertNull( $p->get_attribute( 'width' ) );
+		$p->next_tag( 'div' );
+		$this->assertEquals( 'some-id', $p->get_attribute( 'id' ) );
+		$p->next_tag( 'div' );
+		$this->assertEquals( '100', $p->get_attribute( 'id' ) );
+	}
+
+	/**
+	 * Tests that the `process_directives` does not process the HTML
+	 * inside SVG tags.
+	 *
+	 * @ticket 60517
+	 *
+	 * @covers ::process_directives
+	 */
+	public function test_process_directives_does_not_change_inner_html_in_svgs() {
+		$this->interactivity->state(
+			'myPlugin',
+			array(
+				'id' => 'some-id',
+			)
+		);
+		$html           = '
+			<header>
+				<svg height="100">
```

Comment:
Let's add a `title` tag here just to make sure it works as expected.

```suggestion
				<svg height="100">
					<title>Red Circle</title>
```